### PR TITLE
Update/2.0.0 fix generate slim metadata

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Updated the script to generate slim metadata.. Have tested the metadata with API creation.. works well..

### CODE REVIEW

**Review by holygits — APPROVED** [review]
lgtm.
small nit is it's missing some formatting 
adding a unit test to connect with this metadata should be straightforward? could be a nice improvement for the future

**Comment by KarishmaBothara** [comment]
> lgtm.
> small nit is it's missing some formatting
> adding a unit test to connect with this metadata should be straightforward? could be a nice improvement for the future

Reformatted with prettier.. We have an integration test to check if a meta data option is passed, will it be used and connected fine.. ya but we can have a unit test too in future..

File: scripts/generateSlimMetadata.ts [hunk 1]
```diff
@@ -1,88 +1,85 @@
...
+    'Timestamp',
+    'TransactionPayment',
+    'GenericAsset',
+    'SyloGroups',
+    'SyloE2EE',
+    'SyloDevice',
+    'SyloInbox',
+    'SyloResponse',
+    'SyloVault',
+    'CennzxSpot',
```
Comment (alexsednz) [inline scripts/generateSlimMetadata.ts:29]: Is it still called CennzxSpot or simply Cennzx?

File: scripts/generateSlimMetadata.ts [hunk 2]
```diff
@@ -1,88 +1,85 @@
...
+    } else {
+      // Push an empty module
+      newModules.push({ name: m.name, calls: m.calls, events: m.events });
+    }
+  }
+  let extrinsic = metadata.asLatest.extrinsic;
+  let filteredModule = api.registry.createType('Vec<ModuleMetadataLatest>', newModules);
+  let filteredMetaLatest = api.registry.createType('MetadataLatest', { modules: filteredModule, extrinsic });
+  const filteredMetadataAll = api.registry.createType('MetadataAll', filteredMetaLatest, 12);
+  const mVersionedNew = new MetadataVersioned(api.registry, {
```
Comment (alexsednz) [inline scripts/generateSlimMetadata.ts:50]: Is 12 or 11 the length of KEEP? If so, can you please replace it with KEEP.len?

File: scripts/generateSlimMetadata.ts [hunk 3]
```diff
@@ -1,88 +1,85 @@
...
+      newModules.push(modifiedModule);
+    } else {
+      // Push an empty module
+      newModules.push({ name: m.name, calls: m.calls, events: m.events });
+    }
+  }
+  let extrinsic = metadata.asLatest.extrinsic;
+  let filteredModule = api.registry.createType('Vec<ModuleMetadataLatest>', newModules);
+  let filteredMetaLatest = api.registry.createType('MetadataLatest', { modules: filteredModule, extrinsic });
+  const filteredMetadataAll = api.registry.createType('MetadataAll', filteredMetaLatest, 12);
```
Comment (alexsednz) [inline scripts/generateSlimMetadata.ts:49]: Is 12 or 11 the length of KEEP? If so, can you please replace it with KEEP.len?

File: scripts/generateSlimMetadata.ts (hunk 1 above)
Comment (KarishmaBothara) [reply scripts/generateSlimMetadata.ts:29]: that's right.. we modified it.. thanks

File: scripts/generateSlimMetadata.ts (hunk 3 above)
Comment (KarishmaBothara) [reply scripts/generateSlimMetadata.ts:49]: 11 earlier was the index we were at earlier.. now its 12.. I will trying to search for some generic way like 'latest'.. but unfortunately could not find any..

Comment (KarishmaBothara) [reply scripts/generateSlimMetadata.ts:49]: no..its not length.. its the ENUM index..

**Review by alexsednz — APPROVED** [review]

**Pull request merged**